src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,240 @@
+@click.option("--end-create-time", **END_DATE_DICT)
+@click.option("--begin-update-time", **BEGIN_DATE_DICT)
+@click.option("--end-update-time", **END_DATE_DICT)
+@click.option("--status", help="Filter cases by case status.")
+@format_option
+@sdk_options()
+def _list(
+    state,
+    name,
+    assignee,
+    subject,
+    begin_create_time,
+    end_create_time,
+    begin_update_time,
+    end_update_time,
+    status,
+    format,
+):
+    """List all the cases."""
+    pages = state.sdk.cases.get_all(
+        name=name,
+        assignee=assignee,
+        subject=subject,
+        min_create_time=begin_create_time,
+        max_create_time=end_create_time,
+        min_update_time=begin_update_time,
+        max_update_time=end_update_time,
+        status=status,
+    )
+    formatter = OutputFormatter(format, _get_cases_header())
+    cases = [case for page in pages for case in page["cases"]]
+    if cases:
+        formatter.echo_formatted_list(cases)
+    else:
+        click.echo("No cases found.")
+
+
+def _display_file_events(sdk, case_number):
+    click.echo("\nFile Events:\n")
+    response = sdk.cases.file_events.get_all(case_number)
+    if not response["events"]:
+        click.echo("No events found.")
+    click.echo(pformat(json.loads(response.text)))
+
+
+@cases.command()
+@case_number_arg
+@click.option(
+    "--include-file-events", is_flag=True, help="View events associated to the case."
+)
+@sdk_options()
+@format_option
+def show(state, case_number, format, include_file_events):
+    """Show case details."""
+    formatter = OutputFormatter(format, _get_cases_header())
+    try:
+        response = state.sdk.cases.get(case_number)
+        formatter.echo_formatted_list([response.data])
+        if include_file_events:
+            _display_file_events(state.sdk, case_number)
+    except Py42NotFoundError:
+        click.echo("Invalid case-number {}.".format(case_number))
+
+
+@cases.command()
+@case_number_arg
+@click.option(
+    "--path", type=str, help="File path. Defaults to current directory.", default="."
+)
+@sdk_options()
+def export(state, case_number, path):
+    """Download case detail summary in a pdf file at the given path with name <case_number>_case_summary.pdf."""
+    response = state.sdk.cases.export_summary(case_number)
+    file = os.path.join(path, "{}_case_summary.pdf".format(case_number))
+    with open(file, "wb") as f:
+        f.write(response.content)
+
+
+@cases.group(cls=OrderedGroup)
+@sdk_options()
+def file_events(state):
+    """Fetch file events associated with the case."""
+    pass
+
+
+@file_events.command("list")
+@case_number_arg
+@sdk_options()
+@format_option
+def file_events_list(state, case_number, format):
+    """List all the events associated with the case."""
+    formatter = OutputFormatter(format, _get_events_header())
+    try:
+        response = state.sdk.cases.file_events.get_all(case_number)
+    except Py42NotFoundError as err:
```

Comment:
Change this to `"raise Code42CLIError("Invalid case-number.")`"

Reply:
After the custom exceptions merge in py42, you can dome something even easier:  In `click_ext.groups`, line 54ish, there is a list of imported py42 custom exceptions that automatically get raised/ printed... All we need to do is add any cases-related custom exceptions there and you don't need to do any error handling in the command itself.

However, since that is not merged yet, simply raising a Code42CLIError is the same thing.

Reply:
Done.